orchagent/portsorch.cpp

```diff
@@ -5009,35 +5060,67 @@ void PortsOrch::generateQueueMap()
     {
         if (it.second.m_type == Port::PHY)
         {
-            generateQueueMapPerPort(it.second);
+            generateQueueMapPerPort(it.second, false);
+            if (gMySwitchType == "voq")
+	    {
+                generateQueueMapPerPort(it.second, true);
+	    }
         }
+
+	if (it.second.m_type == Port::SYSTEM)
+	{
+            generateQueueMapPerPort(it.second, true);
+	}
     }
 
     m_isQueueMapGenerated = true;
 }
 
-void PortsOrch::generateQueueMapPerPort(const Port& port)
+void PortsOrch::generateQueueMapPerPort(const Port& port, bool voq)
```

Comment:
instead of passing argument, can we generalize this function generateQueueMapPerPort to call initializeVoqs if global switch type is voq